controllers/storagecluster/cephcluster.go

```diff
@@ -398,16 +398,27 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, s
 			},
 			Resources: newCephDaemonResources(sc),
 			ContinueUpgradeAfterChecksEvenIfNotHealthy: true,
-			LogCollector: rookCephv1.LogCollectorSpec{
-				Enabled:     true,
-				Periodicity: "24h",
-			},
+			LogCollector: sc.Spec.LogCollector,
 			Labels: rookCephv1.LabelsSpec{
 				rookCephv1.KeyMonitoring: getCephClusterMonitoringLabels(*sc),
 			},
 		},
 	}
 
+	// if logCollector is disabled use default values
+	if !cephCluster.Spec.LogCollector.Enabled {
```

Comment:
If it's nullable, the pattern could be to first set the default, then adjust if settings are specified:
```suggestion
	cephCluster.Spec.LogCollector.MaxLogSize = resource.NewQuantity("500Mi", resource.BinarySI)
	if cephCluster.Spec.LogCollector != nil {
```